cli/commands/run-all/command.go

```diff
@@ -0,0 +1,75 @@
+package runall
+
+import (
+	"sort"
+
+	awsproviderpatch "github.com/gruntwork-io/terragrunt/cli/commands/aws-provider-patch"
+	graphdependencies "github.com/gruntwork-io/terragrunt/cli/commands/graph-dependencies"
+	"github.com/gruntwork-io/terragrunt/cli/commands/hclfmt"
+	renderjson "github.com/gruntwork-io/terragrunt/cli/commands/render-json"
+	"github.com/gruntwork-io/terragrunt/cli/commands/terraform"
+	terragruntinfo "github.com/gruntwork-io/terragrunt/cli/commands/terragrunt-info"
+	validateinputs "github.com/gruntwork-io/terragrunt/cli/commands/validate-inputs"
+	"github.com/gruntwork-io/terragrunt/cli/flags"
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/gruntwork-io/terragrunt/pkg/cli"
+)
+
+const (
+	CommandName = "run-all"
+)
+
+var (
+	TerragruntFlagNames = flags.CommonFlagNames
+)
+
+func NewCommand(opts *options.TerragruntOptions) *cli.Command {
+	return &cli.Command{
+		Name:        CommandName,
+		Usage:       "Run a terraform command against a 'stack' by running the specified command in each subfolder.",
+		Description: "The command will recursively find terragrunt modules in the current directory tree and run the terraform command in dependency order (unless the command is destroy, in which case the command is run in reverse dependency order).",
+		Subcommands: newSubCommands(opts),
+		Flags:       flags.NewFlags(opts).Filter(TerragruntFlagNames),
+		Before:      func(ctx *cli.Context) error { return ctx.App.Before(ctx) },
+		Action:      action(opts),
+	}
+}
+
+func action(opts *options.TerragruntOptions) func(ctx *cli.Context) error {
+	return func(ctx *cli.Context) error {
+		opts.RunTerragrunt = func(opts *options.TerragruntOptions) error {
+			if cmd := ctx.Command.Subcommand(opts.TerraformCommand); cmd != nil {
+				ctx := ctx.WithValue(options.ContextKey, opts)
+
+				return cmd.Action(ctx)
+			}
+
+			return terraform.Run(opts)
+		}
+
+		return Run(opts.OptionsFromContext(ctx))
+	}
+}
+
+func newSubCommands(opts *options.TerragruntOptions) cli.Commands {
+	cmds := cli.Commands{
+		terragruntinfo.NewCommand(opts),    // terragrunt-info
+		validateinputs.NewCommand(opts),    // validate-inputs
+		graphdependencies.NewCommand(opts), // graph-dependencies
+		hclfmt.NewCommand(opts),            // hclfmt
+		renderjson.NewCommand(opts),        // render-json
+		awsproviderpatch.NewCommand(opts),  // aws-provider-patch
+	}
+
+	sort.Sort(cmds)
+
+	// add terraform command `*` after sorting to put the command at the end of the list in the help.
+	cmds.Add(terraform.NewCommand(opts))
```

Comment:
This logic looks very similar to what's in `app.go`... Should it be shared somehow?

Reply:
We need add one more command after sorting and there is no `run-all` command in this list as in `app.go`
Unfortunately, any attempt to share this logic only worsens the readability of the code or creates a cycle.